src/typecheck.cpp

```diff
@@ -293,9 +293,10 @@ void print_information(TypedAST::Declaration* ast, TypeChecker& tc) {
 #if DEBUG
 	auto poly = ast->m_decl_type;
 	auto& poly_data = tc.m_core.poly_data[poly];
-	std::cerr << "@@ Type of local variable " << ast->identifier_text() << '\n';
-	std::cerr << "@@ Has " << poly_data.vars.size() << " variables\n";
-	std::cerr << "@@ It is equal to:\n";
+	Log::info(
+	    "Type of local variable '" + ast->identifier_text().str() + "' has " +
+	    std::to_string(poly_data.vars.size()) + " type variables");
```

Comment:
A general purpose logging stream will be the next iteration of the logging system probably.

Reply:
Yeah, we probably want a `BasicLogStream` or something that the other ones inherit from. (future, lol)